### ISSUE TITLE
Build assets which should be served on a subdomain

### ISSUE THREAD

**Issue by brunoscopelliti** [tkareine/jekyll-minibundle#8]
Hi there,

I would like to use your plugin for the assets of my blog, but i'm having some troubles... I think I've tried everything, but at the end I didn't solve my issue, and start thinking this isn't really my fault, but the plugin doesn't hanlde my use case (that is strange, since imho it's a pretty common use case).

Let's say my blog is `example.com`.
I would like to load my assets from a third level subdomain, at which in the future a cdn will respond, eg. `cdn.example.com`.

I'm using the plugin in this way:

```
<link href='{{ site.cdn_url }}/{% ministamp _assets/index.css css/index.css %}'>
```

When I build the site I get the following markup

```
<link href='//cdn.example.com/css/index-abc123.css'>
```

But the file is created in the wrong path, that is, not in `dist/cdn/css/index-abc123.css`, but simply  `dist/css/index-abc123.css`.

In order to make the file be built in the proper folder, I should write

```
<link href='{{ site.cdn_url }}/{% ministamp _assets/index.css cdn/css/index.css %}'>
```

But then the markup is not correct. Note that `cdn` is repeated twice.

```
<link href='//cdn.example.com/cdn/css/index-abc123.css'>
```

Since I noted that {% ministamp %} output is the path passed as second parameter, between the other things I've also tried something like (that would have been an acceptable solution for me):

```
<link href='{{ site.cdn_url }}/{% ministamp _assets/index.css cdn/css/index.css | replace: 'cdn/','/'%}'>
```

But the filter seems to be ignored. I've also tried to assign {% ministamp %} output to another variable, but still nothing.

Currently to make the plugin suitable for my use case I should move the files to the correct destination manually, that isn't properly great. Am I getting wrong something? Does `jekyll-minibundle` support this use case?

Anyway, I would like to thank you for your work on this plugin!

Cheers

**Comment by brunoscopelliti** [comment]
Hi there,

it's me, again :)

I perhaps come with a solution for my use case, that doesn't require to change the code of this plugin :tada:... so I would like to share, in case someone has the same need.

I use the `capture` liquid tag to capture the output of `ministamp`, than use a replace to remove the part of the path I don't need. That's the code:

```
{% capture stylesheet %}{% ministamp _assets/css/index.css resources/marvinjs/css/index.css %}{% endcapture %}
<link href='{{ site.cdn_url }}{{ stylesheet | replace: 'resources/','/' }}' rel='stylesheet' type='text/css'>
```

**Comment by tkareine** [comment]
Hi! I'm happy you found a way with the `capture` Liquid tag! :)

I think I understand your use case, but I'm not sure yet if it falls within the scope of the plugin. Specifically, I'm considering

1. alternative ways to accomplish the use case (there's at least two: moving the asset file to another directory after generating the site, or using Liquid output capture) vs. their awkwardness,

2. nice and general enough syntax to express the use case in the plugin, and

3. implementation complexity.

For the 1st point, Liquid output capture seems fairly ok to me as an alternative. Practically, it just takes an additional line of code in the call site.

For the 2nd point, I was thinking of extending the `ministamp` tag to take in YAML. For example:

```
<link href='{{ site.cdn_url }}{% ministamp {source_path: _assets/css/index.css, destination_path: resources/marvinjs/css/index.css, destination_url: /marvinjs/css/index.css} }}' rel='stylesheet' type='text/css'>
```

That would allow backward compatibility with the old syntax, making this

```
<link href='/{% ministamp _assets/css/index.css css/index.css }}' type='text/css'>
```

to be just a shorthand for:

```
<link href='/{% ministamp {source_path: _assets/css/index.css, destination_path: css/index.css} }}' type='text/css'>
```

For the 3rd point, I believe the implementation will be pretty straightforward, but there's a quite a lot of places that need to be changed. In addition, if we add the feature to `ministamp` tag, it should be done for `minibundle` block, too.

Any thoughts about this? I'll think and play with the code a bit more before deciding what to do.

Unfortunately, I don't have much spare time nowadays, so it will take time for me to reply.

**Comment by brunoscopelliti** [comment]
Thanks for having found the time to write a such detailed answer.

I mostly agree with you on every point.

1 - I agree using the `capture` tag could be considered a fine way to achieve the wanted behaviour (and far way better than moving the file post generation)... 
However in this case I feel we should improve a little the documentation, so the next person who has my same use case won't spend as much time as me to find a way out.
Do you think documenting my use case in the project README is valuable? Eventually I could help with a pull request.

2 - I agree that the solution you suggest is probably the best for the vast majority of people using `ministamp`.
Another possibility I've considered (that breaks backward compatibility) is to change the output of `ministamp`, to make it only returns the filename with the hash (the only information we don't know a priori).
In this case `ministamp` should be used as follows

```
<link href='{{site.cdn}}/css/{% ministamp _assets/css/index.css resources/css/index.css }}' type='text/css'>
<!-- or -->
<link href='/resources/css/{% ministamp _assets/css/index.css resources/css/index.css }}' type='text/css'>
```

3 - I've tried to understand the working of the plugin, but I haven't gone too far (I am not a rubyst)... so, sorry I can't help with the implementation... however I pretty agree, it shouldn't be too much hard since I don't expect `destination_path` is used in a lot of places (I may be tremendously wrong 😄 )

**Comment by tkareine** [comment]
I added an example of using Liquid `capture` block to the readme. Thanks for suggesting it!

So far, I haven't had the chance for drafting the implementation for this use case further. I'll leave the issue open, hoping I'll get an evening or two for this later.

**Comment by tkareine** [comment]
This use case in now supported with the `render_baseurl_only` option of `ministamp` tag. For example:

```liquid
<link href="{{ site.cdn_url }}/marvinjs/css/{% ministamp { source_path: _assets/css/index.css, destination_path: resources/marvinjs/css/index.css, render_basename_only: true } %}" rel="stylesheet">
```

Available in release 2.2.0.
